src/Aspire.Hosting/Health/ResourceHealthCheckService.cs

```diff
@@ -52,11 +52,20 @@ private async Task MonitorResourceHealthAsync(ResourceEvent initialEvent, Cancel
             //       dynamically add health checks at runtime. If this changes then we
             //       would need to revisit this and scan for transitive health checks
             //       on a periodic basis (you wouldn't want to do it on every pass.
-            var resourceReadyEvent = new ResourceReadyEvent(resource, services);
-            await eventing.PublishAsync(
-                resourceReadyEvent,
-                EventDispatchBehavior.NonBlockingSequential,
-                cancellationToken).ConfigureAwait(false);
+            _ = Task.Run(async () =>
```

Comment:
Is no await here anymore ok?

Should there be something to handle logging errors from the unawaited task?

Reply:
The behavior before was non blocking, that means don't block on the event handlers firing.